pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -91,6 +92,53 @@ func ReconcileStatus(
 	return apierrorsutil.NewAggregate(errs)
 }
 
+var anno2ConditionRegex = regexp.MustCompile(`^condition.vmoperator.vmware.com.protected/(.+)?$`)
+
+// reconcileStatusAnno2Conditions sets conditions on the VM based on
+// annotation values.
+func reconcileStatusAnno2Conditions(
+	vmCtx pkgctx.VirtualMachineContext,
+	_ ctrlclient.Client,
+	_ *object.VirtualMachine,
+	_ ReconcileStatusData) []error { //nolint:unparam
+
+	for k, v := range vmCtx.VM.Annotations {
+		if anno2ConditionRegex.MatchString(k) {
```

Comment:
Why the regex with the capture group when not using that, vs `strings.HasPrefix()`?